src/test/java/com/melon/portfoliomanager/UserControllerTest.java

```diff
@@ -0,0 +1,81 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.melon.portfoliomanager.controllers.UserController;
+import com.melon.portfoliomanager.exceptions.NullFieldException;
+import com.melon.portfoliomanager.exceptions.UsernameAlreadyUsedException;
+import com.melon.portfoliomanager.models.User;
+import com.melon.portfoliomanager.repositories.UserRepository;
+import com.melon.portfoliomanager.services.UserService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import java.util.ArrayList;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+public class UserControllerTest {
+
+    @InjectMocks
+    UserController userController;
+
+    @Mock
+    UserService userService;
+
+    ObjectMapper objectMapper = new ObjectMapper();
+
+    MockMvc mockMvc;
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        MockitoAnnotations.initMocks(this);
+        mockMvc = MockMvcBuilders.standaloneSetup(userController).build();
+    }
+
+    @Test
+    public void createUser_SuccessfulCreation_ReturnsCreated() throws Exception {
+        User user = new User("testUser", "test@test.com", "John", "Doe");
+
+        when(userService.createUser(any(User.class))).thenReturn(user);
+
+        mockMvc.perform(post("/users/add")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(user)))
+                .andExpect(status().isCreated());
+    }
+
+    @Test
+    public void createUser_UsernameAlreadyExists_ReturnsBadRequest() throws Exception {
+        User user = new User("testUser", "test@test.com", "John", "Doe");
+
+        when(userService.createUser(any(User.class))).thenThrow(UsernameAlreadyUsedException.class);
+
+        mockMvc.perform(post("/users/add")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(user)))
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    public void createUser_NullField_ReturnsBadRequest() throws Exception {
+        User user = new User(null, "test@test.com", "John", "Doe");
+
+        when(userService.createUser(any(User.class))).thenThrow(NullFieldException.class);
+
+        mockMvc.perform(post("/users/add")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(user)))
+                .andExpect(status().isBadRequest());
+    }
+
+}
```

Comment:
It is best to have a test verifying that if an unexpected error occurs then 500 status code is returned.